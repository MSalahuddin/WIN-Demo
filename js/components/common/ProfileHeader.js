/* eslint-disable no-nested-ternary */
import React, { useContext } from 'react';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import { NavigationActions, StackActions } from 'react-navigation';
import { Platform } from 'react-native'
import Ellipse from './Ellipse';
import { color } from '../../styles';
import IconButton from './IconButton';
import ProfileIconButton from './ProfileIconButton';
import { UserContext } from '../../context/User.context';
import { scale, scaleHeight, scaleWidth } from '../../platformUtils';
import { coin, ticket, redBack, gearBorderless } from '../../../assets/images';
import { Win_demoWIcon } from '../../../assets/svgs';
import { SCREENS } from '../../constants';
const Header =
  styled.SafeAreaView`
  align-items: center;
  flex-direction: row;
  justify-content: space-between;
  margin-horizontal: ${scaleWidth(Platform.OS === 'android' ? 0 : 16)};
  margin-bottom: ${Platform.OS === 'android' ? scaleHeight(10) : scaleHeight(24)};
  margin-top: ${Platform.OS === 'android' ? scaleHeight(27) : scaleHeight(10)};
`;

const BackButton = styled(IconButton)`
  margin-top: 0;
`;

const IconButtonWrapper = styled(IconButton)`
  margin-left: ${Platform.OS === 'android' ? scaleWidth(8) : 0};
  margin-top: 0;
`;
const ProfileIconWrapper = styled(ProfileIconButton)`
margin-left: ${Platform.OS === 'android' ? scaleWidth(8) : 0};
margin-top: 0;
`;


const ProfileHeader = ({ navigation, isProfileImageShown }) => {
  const {
    tokens,
    tickets,
    vipLevel,
    profilePicture,
    isFreePlayHeaderIconNotificationVisible,
    totalFreePlays,
    setFreePlayHeaderNotificationVisible,
    FreePlayStatus,
    liveArcade
  } = useContext(UserContext);

  const vipLevelId = vipLevel?.vipLevelId || 0;
  const imageSource = profilePicture ? { uri: profilePicture } : null;
  const { navigate } = navigation;

  const profileIconOnPress = () => {
    navigate(SCREENS.ACCOUNT_PROFILE);
  };

  const renderIcon = () => {
    return (
      <Win_demoWIcon
        width={Platform.OS === 'android' ? 25 : 30}
        height={Platform.OS === 'android' ? 25 : 30}
      />
    )
  }

  const renderFirstIcon = () => {
    const routeName = navigation?.state?.routeName;
    if (isProfileImageShown) {
      return (
        <IconButtonWrapper
          size={Platform.OS === 'android' ? 25 : 30}
          icon={renderIcon}
          backGroundColor={color.governorBay}
          borderRadius={scale(50)}
          paddingVertical={scale(8)}
          paddingHorizontal={scale(8)}
          onPress={() => {

            const resetAction = routeName === SCREENS.DIGITAL_GAME_ROOM ?
              StackActions.reset({
                index: 0,
                actions: [NavigationActions.navigate({ routeName: SCREENS.GAME_ROOM })]
              }) : StackActions.reset({
                index: 0,
                actions: [NavigationActions.navigate({ routeName: SCREENS.DIGITAL_GAME_ROOM })]
              });
            navigation.dispatch(resetAction);

            const navigate = StackActions.reset({
              index: 0,
              actions: [NavigationActions.navigate({ routeName: SCREENS.DIGITAL_GAME_ROOM })]
            });
            navigation.dispatch(liveArcade ? resetAction : navigate);

          }}
          isSvg
        />
      );
    }
    return (
      <BackButton
        testID="back-button"
        onPress={() => {
          navigation.goBack();
        }}
        icon={redBack}
      />
    );
  };
  const handleOnPress = () => {
    navigate(SCREENS.GAME_CARD_RELOAD)
    setFreePlayHeaderNotificationVisible(false)
  }
  return (
    <>
      <Header>
        {renderFirstIcon()}
        {Platform.OS === "android" ? <Ellipse
          testID="coin-button"
          width={!vipLevel ? scaleWidth(110) : scaleWidth(93)}
          borderRadius={44}
          icon={coin}
          amount={tokens}
          freePlay={totalFreePlays}
          freePlayNotification={isFreePlayHeaderIconNotificationVisible}
          isPlusPresent
          onPress={handleOnPress}
          freePlayStatus={FreePlayStatus}
        />
          :
          <Ellipse
            testID="coin-button"
            width={95}
            borderRadius={44}
            icon={coin}
            amount={tokens}
            freePlay={totalFreePlays}
            freePlayNotification={isFreePlayHeaderIconNotificationVisible}
            isPlusPresent
            onPress={handleOnPress}
            freePlayStatus={FreePlayStatus}
          />}
        {Platform.OS === "android" ?
          <Ellipse
            testID="ticket-button"
            width={!vipLevel ? scaleWidth(100) : scaleWidth(85)}
            borderRadius={44}
            icon={ticket}
            amount={tickets}
            onPress={() => {
              navigate(SCREENS.WINNERS_CIRCLE);
            }}
          />
          : <Ellipse
            testID="ticket-button"
            width={88}
            borderRadius={44}
            icon={ticket}
            amount={tickets}
            onPress={() => {
              navigate(SCREENS.WINNERS_CIRCLE);
            }}
          />}

        {!vipLevel ?
          (
            <>
              {imageSource && <ProfileIconWrapper size={40} icon={imageSource} onPress={() => profileIconOnPress()} />}
              {!imageSource &&
                <ProfileIconWrapper size={40} icon={gearBorderless} onPress={() => profileIconOnPress()} />}
            </>
          )
          : (
            <Ellipse
              width={Platform.OS === 'android' ? scaleWidth(76) : 88}
              borderRadius={44}
              icon={imageSource}
              vipLevelId={vipLevelId}
              isIconAlignedRight
              onPress={() => {
                profileIconOnPress();
              }}
            />
          )}
      </Header>
    </>
  );
};

ProfileHeader.propTypes = {
  isProfileImageShown: PropTypes.bool,
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
    goBack: PropTypes.func.isRequired,
    dispatch: PropTypes.func.isRequired
  }).isRequired
};

ProfileHeader.defaultProps = {
  isProfileImageShown: false
};

export default ProfileHeader;
