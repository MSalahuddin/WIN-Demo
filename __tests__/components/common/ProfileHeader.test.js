import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import MockProvider from '../../MockProvider';
import ProfileHeader from '../../../js/components/common/ProfileHeader';

describe('Component ProfileHeader', () => {
  const navigation = { dispatch: jest.fn(), navigate: jest.fn(), goBack: jest.fn() };

  test('renders component with profile image correctly', () => {
    const output = render(
      <MockProvider>
        <ProfileHeader navigation={navigation} isProfileImageShown />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('renders component when isUserLoggedIn', () => {
    const output = render(
      <MockProvider isUserLoggedIn>
        <ProfileHeader navigation={navigation} isProfileImageShown />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('renders component with back button correctly', () => {
    const output = render(
      <MockProvider>
        <ProfileHeader navigation={navigation} />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('calls goBack when profile icon is pressed', () => {
    const { getByTestId } = render(
      <MockProvider>
        <ProfileHeader navigation={navigation} />
      </MockProvider>
    );
    fireEvent(getByTestId('back-button'), 'press');
    expect(navigation.goBack).toHaveBeenCalled();
  });

  test('calls navigate when coin icon is pressed', () => {
    const { getByTestId } = render(
      <MockProvider>
        <ProfileHeader navigation={navigation} />
      </MockProvider>
    );
    fireEvent(getByTestId('coin-button'), 'press');
    expect(navigation.navigate).toHaveBeenCalled();
  });

  test('calls navigate when ticket is pressed', () => {
    const { getByTestId } = render(
      <MockProvider>
        <ProfileHeader navigation={navigation} />
      </MockProvider>
    );
    fireEvent(getByTestId('ticket-button'), 'press');
    expect(navigation.navigate).toHaveBeenCalled();
  });
});
