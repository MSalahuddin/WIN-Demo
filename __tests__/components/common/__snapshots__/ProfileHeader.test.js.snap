// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ProfileHeader renders component when isUserLoggedIn 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 48,
        "marginHorizontal": 32,
        "marginTop": 20,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": 0,
        "marginTop": 0,
        "opacity": 1,
      }
    }
    testID="icon-button"
  >
    <Image
      opacity={1}
      resizeMode="contain"
      size={40}
      source={1}
      style={
        Array [
          Object {
            "height": 60,
            "width": 60,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 44,
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "center",
        "opacity": 1,
        "width": 142.5,
      }
    }
    testID="coin-button"
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "height": 36,
            "marginRight": 7.5,
            "width": 36,
          },
        ]
      }
    />
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#000000"
      fontFamily="MEDIUM"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Calibre-Medium",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 10.5,
          },
        ]
      }
    >
      0
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#72D59E",
          "borderRadius": 13.5,
          "height": 27,
          "opacity": 1,
          "paddingLeft": 6,
          "paddingTop": 3,
          "position": "absolute",
          "right": -6.75,
          "width": 27,
        }
      }
      testID="coin-button"
    >
      <Text
        alignCenter={false}
        allowFontScaling={false}
        color="#FFFFFF"
        fontFamily="MEDIUM"
        isUnderlined={false}
        isUppercase={false}
        maxFontSizeMultiplier={1.3}
        size="NORMAL"
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Calibre-Medium",
              "fontSize": 30,
              "justifyContent": "center",
              "textAlign": "left",
              "textDecorationColor": "#FFFFFF",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
              "textTransform": "none",
            },
          ]
        }
      >
        +
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 44,
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "center",
        "opacity": 1,
        "width": 132,
      }
    }
    testID="ticket-button"
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "height": 36,
            "marginRight": 7.5,
            "width": 36,
          },
        ]
      }
    />
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#000000"
      fontFamily="MEDIUM"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Calibre-Medium",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 10.5,
          },
        ]
      }
    >
      0
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 44,
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "center",
        "opacity": 1,
        "width": 132,
      }
    }
    testID="ellipse-button"
  >
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#000000"
      fontFamily="MEDIUM"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Calibre-Medium",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 10.5,
          },
        ]
      }
    >
      VIP 1
    </Text>
    <Image
      resizeMode="contain"
      source={null}
      style={
        Array [
          Object {
            "borderRadius": 225,
            "height": 36,
            "marginLeft": 7.5,
            "width": 36,
          },
        ]
      }
    />
  </View>
</RCTSafeAreaView>
`;

exports[`Component ProfileHeader renders component with back button correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 48,
        "marginHorizontal": 32,
        "marginTop": 20,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": 24,
        "marginTop": 0,
        "opacity": 1,
      }
    }
    testID="back-button"
  >
    <Image
      opacity={1}
      resizeMode="contain"
      size={40}
      source={1}
      style={
        Array [
          Object {
            "height": 60,
            "width": 60,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 44,
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "center",
        "opacity": 1,
        "width": 142.5,
      }
    }
    testID="coin-button"
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "height": 36,
            "marginRight": 7.5,
            "width": 36,
          },
        ]
      }
    />
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#000000"
      fontFamily="MEDIUM"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Calibre-Medium",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 10.5,
          },
        ]
      }
    >
      0
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#72D59E",
          "borderRadius": 13.5,
          "height": 27,
          "opacity": 1,
          "paddingLeft": 6,
          "paddingTop": 3,
          "position": "absolute",
          "right": -6.75,
          "width": 27,
        }
      }
      testID="coin-button"
    >
      <Text
        alignCenter={false}
        allowFontScaling={false}
        color="#FFFFFF"
        fontFamily="MEDIUM"
        isUnderlined={false}
        isUppercase={false}
        maxFontSizeMultiplier={1.3}
        size="NORMAL"
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Calibre-Medium",
              "fontSize": 30,
              "justifyContent": "center",
              "textAlign": "left",
              "textDecorationColor": "#FFFFFF",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
              "textTransform": "none",
            },
          ]
        }
      >
        +
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 44,
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "center",
        "opacity": 1,
        "width": 132,
      }
    }
    testID="ticket-button"
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "height": 36,
            "marginRight": 7.5,
            "width": 36,
          },
        ]
      }
    />
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#000000"
      fontFamily="MEDIUM"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Calibre-Medium",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 10.5,
          },
        ]
      }
    >
      0
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 44,
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "center",
        "opacity": 1,
        "width": 132,
      }
    }
    testID="ellipse-button"
  >
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#000000"
      fontFamily="MEDIUM"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Calibre-Medium",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 10.5,
          },
        ]
      }
    >
      VIP 1
    </Text>
    <Image
      resizeMode="contain"
      source={null}
      style={
        Array [
          Object {
            "borderRadius": 225,
            "height": 36,
            "marginLeft": 7.5,
            "width": 36,
          },
        ]
      }
    />
  </View>
</RCTSafeAreaView>
`;

exports[`Component ProfileHeader renders component with profile image correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 48,
        "marginHorizontal": 32,
        "marginTop": 20,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": 0,
        "marginTop": 0,
        "opacity": 1,
      }
    }
    testID="icon-button"
  >
    <Image
      opacity={1}
      resizeMode="contain"
      size={40}
      source={1}
      style={
        Array [
          Object {
            "height": 60,
            "width": 60,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 44,
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "center",
        "opacity": 1,
        "width": 142.5,
      }
    }
    testID="coin-button"
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "height": 36,
            "marginRight": 7.5,
            "width": 36,
          },
        ]
      }
    />
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#000000"
      fontFamily="MEDIUM"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Calibre-Medium",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 10.5,
          },
        ]
      }
    >
      0
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#72D59E",
          "borderRadius": 13.5,
          "height": 27,
          "opacity": 1,
          "paddingLeft": 6,
          "paddingTop": 3,
          "position": "absolute",
          "right": -6.75,
          "width": 27,
        }
      }
      testID="coin-button"
    >
      <Text
        alignCenter={false}
        allowFontScaling={false}
        color="#FFFFFF"
        fontFamily="MEDIUM"
        isUnderlined={false}
        isUppercase={false}
        maxFontSizeMultiplier={1.3}
        size="NORMAL"
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Calibre-Medium",
              "fontSize": 30,
              "justifyContent": "center",
              "textAlign": "left",
              "textDecorationColor": "#FFFFFF",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
              "textTransform": "none",
            },
          ]
        }
      >
        +
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 44,
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "center",
        "opacity": 1,
        "width": 132,
      }
    }
    testID="ticket-button"
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "height": 36,
            "marginRight": 7.5,
            "width": 36,
          },
        ]
      }
    />
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#000000"
      fontFamily="MEDIUM"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Calibre-Medium",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 10.5,
          },
        ]
      }
    >
      0
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 44,
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "center",
        "opacity": 1,
        "width": 132,
      }
    }
    testID="ellipse-button"
  >
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#000000"
      fontFamily="MEDIUM"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Calibre-Medium",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 10.5,
          },
        ]
      }
    >
      VIP 1
    </Text>
    <Image
      resizeMode="contain"
      source={null}
      style={
        Array [
          Object {
            "borderRadius": 225,
            "height": 36,
            "marginLeft": 7.5,
            "width": 36,
          },
        ]
      }
    />
  </View>
</RCTSafeAreaView>
`;
